### DISCUSSION TITLE
Documentation about all monitored sensors?

### DISCUSSION THREAD

**Discussion by nicx** [Arnold-n/P1P2MQTT#75] [General]
Hey, is there any documentations about the sensors in detail? I am completely new in this "Daikin business" and have no clue about the meanings of the sensors. :) 

for example:
- what ist the "M9_Power_Gasboiler" for? why does it have negative only values?
- why do I have a "gasboiler" with all that gas boiler sensors at all? I am using no hybrid/gas heating.
- what is "T1_Temperature_Refrigerant_3_Q"?
- what are the unknown sensors for?
- etcpp. :)

here is a complete list of my sensors. some of them are self explaining, most not (for me):

```
A8_Control_ID
"0xF0"
 
A8_Error_Budget
19
 
A8_Error_Oversized_Packet_Count
0
 
A8_Error_Read_Count
5
 
A8_Error_Read_Type
8
 
A8_Writes_Refused
0
 
A9_ESP_Output_Mode
"0x00013003"
 
A9_MQTT_disconnected_skipped_packets
0 events
 
A9_MQTT_disconnected_time
0 s
 
A9_MQTT_disconnected_time_total
0 s
 
A9_MQTT_messages_skipped_not_connected
0 events
 
C1_Counter_Gasboiler_DHW
0
 
C1_Counter_Gasboiler_Heating
0
 
C1_Counter_Gasboiler_Total
0
 
C1_Counter_Unknown_00
0
 
C1_Counter_Unknown_01
0
 
C1_Counter_Unknown_02
0
 
C1_Electricity_Consumed_Backup_DHW
0 kWh
 
C1_Electricity_Consumed_Backup_Heating
0 kWh
 
C1_Electricity_Consumed_Compressor_Cooling
0 kWh
 
C1_Electricity_Consumed_Compressor_DHW
3 kWh
 
C1_Electricity_Consumed_Compressor_Heating
0 kWh
 
C1_Electricity_Consumed_Total
3 kWh
 
C1_Energy_Produced_Heatpump_Cooling
0 kWh
 
C1_Energy_Produced_Heatpump_DHW
14 kWh
 
C1_Energy_Produced_Heatpump_Heating
0 kWh
 
C1_Energy_Produced_Heatpump_Total
14 kWh
 
C1_Hours_Backup1_DHW
0 hours
 
C1_Hours_Backup1_Heating
0 hours
 
C1_Hours_Backup2_DHW
0 hours
 
C1_Hours_Backup2_Heating
0 hours
 
C1_Hours_Circulation_Pump
2 hours
 
C1_Hours_Compressor_Cooling
0 hours
 
C1_Hours_Compressor_DHW
1 hours
 
C1_Hours_Compressor_Heating
0 hours
 
C1_Hours_Gasboiler_DHW
0 hours
 
C1_Hours_Gasboiler_Heating
0 hours
 
C1_Hours_Unknown_34
0 hours
 
C1_Hours_Unknown_35
0 hours
 
C1_Starts_Compressor
4
 
C1_Starts_Gasboiler
0
 
M1_Flow
0,0 l/min
 
M9_Power_Gasboiler
-0,00 kW
 
M9_Power_Heatpump
0,00 kW
 
S0_DHW_Booster_Active
0
 
S0_DHW_OnOff
1
 
S0_Heating_Cooling_Auto
0
 
S0_Heating_modus
80
 
S0_Heating_OnOff
1
 
S0_Quiet_Mode
0
 
S0_Target_Temperature_Room
20,0 °C
 
S0_Temperature_Setpoint_LWT_Cooling_Zone_Add
20,0 °C
 
S0_Temperature_Setpoint_LWT_Cooling_Zone_Main
20,0 °C
 
S0_Temperature_Setpoint_LWT_Heating_Zone_Add
25,0 °C
 
S0_Temperature_Setpoint_LWT_Heating_Zone_Main
25,0 °C
 
S1_Circulation_Pump_OnOff
0
 
S1_Compressor_OnOff
0
 
S1_Cooling_OnOff_1
0
 
S1_Defrost_Operation
0
 
S1_DHW_Active_1
0
 
S1_DHW_Mode_OnOff
0
 
S1_DHW_Setpoint_Response
45,0 °C
 
S1_Gasboiler_Active_0
0
 
S1_Gasboiler_On
0
 
S1_Heating_OnOff_1
1
 
S1_Heatpump_On
1
 
S1_Preferential_Mode
0
 
S1_Target_Temperature_DHW
45,0 °C
 
S1_Valve_DHW_Tank
1
 
S1_Valve_Zone_Main
0
 
T0_Temperature_Room
20,0 °C
 
T1_Target_LWT_Zone_Add
25,0 °C
 
T1_Target_LWT_Zone_Main
25,0 °C
 
T1_Temperature_DHW_Tank
45,5 °C
 
T1_Temperature_HP2Gas_Water
45,8 °C
 
T1_Temperature_Leaving_Water
42,0 °C
 
T1_Temperature_Outside_1
18,5 °C
 
T1_Temperature_Outside_2
21,0 °C
 
T1_Temperature_Refrigerant_1
38,7 °C
 
T1_Temperature_Refrigerant_2
18,0 °C
 
T1_Temperature_Refrigerant_3_Q
0,0 °C
 
T1_Temperature_Return_Water
41,4 °C
 
T1_Temperature_Room
20,0 °C
 
T1_Temperature_Unused_400011_16
0,0 °C
 
T1_Temperature_Unused_400011_18
-128,0 °C
 
T1_Temperature_Unused_400015_0
43,1 °C
```

**Comment by nicx** [comment]
@Arnold-n can you help me? Is there any documentation (at least for my concrete questions)?

in addition, I wonder if I can somehow get the current electrical power consumption out and not just the consumed power. as far as I interpret the sensor "M9_Power_Heatpump" this is not the electrical power, but the "generating" power of my outside altherma (because I do have a 8kW model and the sensor is 8kW in peak).

**Comment by Arnold-n** [comment]
Fair question, the current documentation focuses more on the P1/P2 raw data than on the MQTT output. A combination of code and protocol gives some answers, but I see a clear need for a separate documentation file here.

To answer a few questions:
-M9_Power_Gasboiler shows the gas heating output for hybrid systems. It is calculated based on water flow and a temperature difference between 2 sensors; in some other systems  this gives a slightly negative or a totally wrong answer. The current code does not yet change the output based on the Altherma model,
-M9_Power_Heatpump is  real-time estimate for generated heat based on water flow (some systems do not measure this) and temperature difference between return water temperature and water temperature leaving the heat pump, and
-for the unknown sensors: we do not know yet... some are always zero, some have a value that we couldn't pinpoint yet.

Electricity consumption is not output directly by Daikin systems; instead a low-resolution kWh counter is updated once every hour or once every two hours.

Generated heat is also not output directly by Daikin either; instead a kWh counter is updated whenever a new kWh of heat has been generated. 

These counters are only output by P1P2MQTT if counter requests are switched on ("C2" command).
